2008-08-18  Aaron W. LaFramboise  <aaronavay62@aaronwl.com>

	* sysdep/i386/backtrace.h (DONT_REFERENCE_MAIN) [_WIN32]: New.
	(fallback_backtrace) [DONT_REFERENCE_MAIN]: Don't generate references to main().

Index: libjava/sysdep/i386/backtrace.h
===================================================================
--- a/libjava/sysdep/i386/backtrace.h	(revision 139186)
+++ b/libjava/sysdep/i386/backtrace.h	(working copy)
@@ -13,7 +13,13 @@ details.  */
 
 #include <java-stack.h>
 
+#ifdef _WIN32
+#define DONT_REFERENCE_MAIN
+#endif
+
+#ifndef DONT_REFERENCE_MAIN
 extern int main (int, char **);
+#endif
 
 /* The context used to keep track of our position while unwinding through
    the call stack.  */
@@ -104,7 +110,10 @@ fallback_backtrace (_Unwind_Trace_Fn tra
                             const char **, bool))_Jv_RunMain;
       if (ctx.meth_addr == (_Jv_uintptr_t)jv_runmain
           || ctx.meth_addr == (_Jv_uintptr_t)_Jv_ThreadStart
-          || (ctx.meth_addr - (_Jv_uintptr_t)main) < 16)
+#ifndef DONT_REFERENCE_MAIN
+          || (ctx.meth_addr - (_Jv_uintptr_t)main) < 16
+#endif
+          )
         break;
     }
 
